ui/app/components/auth-jwt.js

```diff
@@ -28,57 +26,50 @@ export default Component.extend({
   roleName: null,
   role: null,
   errorMessage: null,
+  isOIDC: true,
+
   onRoleName() {},
   onLoading() {},
   onError() {},
   onNamespace() {},
 
   didReceiveAttrs() {
-    this._super();
-    const debounce = !this.oldSelectedAuthPath && !this.selectedAuthPath;
-
-    if (this.oldSelectedAuthPath !== this.selectedAuthPath || debounce) {
-      this.fetchRole.perform(this.roleName, { debounce });
+    this._super(...arguments);
+    // if mount path or type changes we need to check again for JWT configuration
+    const didChangePath = this._authPath !== this.selectedAuthPath;
+    const didChangeType = this._authType !== this.selectedAuthType;
+
+    if (didChangePath || didChangeType) {
+      // path updates as the user types so we need to debounce that event
+      const wait = didChangePath ? 500 : 0;
+      debounce(this, 'fetchRole', wait);
     }
-
-    this.set('errorMessage', null);
-    this.set('oldSelectedAuthPath', this.selectedAuthPath);
+    this._authPath = this.selectedAuthPath;
+    this._authType = this.selectedAuthType;
```

Comment:
Since this isn't glimmerized, do we need to do `this.set('_authPath', this.selectedAuthPath)` and `this.set('_authType_', this.selectedAuthType)`?

Reply:
Not in this case since we aren't accessing the property in the template nor are we using it as a dependent property for any computed properties. Basically we don't need any change detection to be triggered when updating these values.

Reply:
makes sense to me thanks for explaining!